monomer_test.go

```diff
@@ -1,19 +1,190 @@
 package monomer_test
 
 import (
+	"math/big"
+	"strconv"
 	"testing"
+	"time"
 
+	"github.com/cometbft/cometbft/types"
+	opeth "github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/polymerdao/monomer"
 	"github.com/polymerdao/monomer/testutils"
 	"github.com/stretchr/testify/require"
+	"github.com/test-go/testify/assert"
 )
 
-func TestBlockToEth(t *testing.T) {
-	l1InfoTx, depositTx, cosmosEthTx := testutils.GenerateEthTxs(t)
-	block := testutils.GenerateBlockFromEthTxs(t, l1InfoTx, []*ethtypes.Transaction{depositTx}, []*ethtypes.Transaction{cosmosEthTx})
-	ethBlock, err := block.ToEth()
-	require.NoError(t, err)
-	for i, ethTx := range []*ethtypes.Transaction{l1InfoTx, depositTx, cosmosEthTx} {
-		require.EqualExportedValues(t, ethTx, ethBlock.Body().Transactions[i])
+func TestChainID(t *testing.T) {
+	n := int64(12345)
+	id := monomer.ChainID(n)
+
+	idStr := strconv.FormatInt(n, 10)
+	assert.Equal(t, idStr, id.String())
+
+	bigInt := big.NewInt(n)
+	assert.Equal(t, bigInt, id.Big())
+
+	hexID := (*hexutil.Big)(bigInt)
+	assert.Equal(t, hexID, id.HexBig())
+}
+
+func newTestHeader() *monomer.Header {
+	return &monomer.Header{
+		ChainID:    12345,
+		Height:     67890,
+		Time:       uint64(time.Now().Unix()),
+		StateRoot:  common.HexToHash("0x1"),
+		ParentHash: common.HexToHash("0x2"),
+		GasLimit:   3000000,
+		Hash:       common.HexToHash("0x3"),
 	}
 }
+
+func TestHeader(t *testing.T) {
+	timestamp := time.Now()
+	header := newTestHeader()
+
+	t.Run("ToComet", func(t *testing.T) {
```

Comment:
Let's have separate tests for these cases. That way we don't share the `header` and `timestamp` vars